File: scripts/check_braces.py
import pathlib
path = pathlib.Path(r"c:\Users\mylig\Github\FOMO Life\unit-tests\App.test.js")
lines = path.read_text(encoding='utf-8').splitlines()
balance = 0
for i, line in enumerate(lines, start=1):
    for ch in line:
        if ch == '{':
            balance += 1
        elif ch == '}':
            balance -= 1
    if balance < 0:
        print('negative at', i)
        balance = 0
    if balance != 0:
        print(f"line {i} balance {balance} {line.strip()}")
# now output balance at end
print('final balance', balance)

File: scripts/check_braces.js
const fs = require('fs');
const path = 'unit-tests/App.test.js';
const text = fs.readFileSync(path, 'utf8');
const lines = text.split(/\r?\n/);
let bal = 0;
for (let i = 0; i < lines.length; i++) {
  let line = lines[i];
  for (let j = 0; j < line.length; j++) {
    const ch = line[j];
    if (ch === '{') {
      bal++;
      console.log(`line ${i+1} col ${j+1} ++ balance=${bal}`);
    } else if (ch === '}') {
      bal--;
      console.log(`line ${i+1} col ${j+1} -- balance=${bal}`);
    }
  }
}
console.log('final balance', bal);
